tests/admin.test.ts

```diff
@@ -182,3 +185,193 @@ describe('bonus points submission', () => {
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        accessUpdates: [
+          { user: staffUser.email, accessType: UserAccessType.MERCH_STORE_MANAGER },
+          { user: standardUser.email, accessType: UserAccessType.MARKETING },
+          { user: marketingUser.email, accessType: UserAccessType.MERCH_STORE_DISTRIBUTOR },
+          { user: merchStoreDistributorUser.email, accessType: UserAccessType.STAFF },
+        ],
+      }, standard);
+    }).rejects.toThrow(ForbiddenError);
+  });
+
+  test('attempt to update duplicate users', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+
+    const userOne = UserFactory.fake({ accessType: UserAccessType.STAFF, email: 'smhariha@ucsd.edu' });
+    // const userTwo = UserFactory.fake({ accessType: UserAccessType.STANDARD, email: 'smhariha@ucsd.edu' });
+
+    await new PortalState()
+      .createUsers(userOne, admin)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        accessUpdates: [
+          { user: userOne.email, accessType: UserAccessType.MERCH_STORE_MANAGER },
+          { user: userOne.email, accessType: UserAccessType.STAFF },
+        ],
+      }, admin);
+    }).rejects.toThrow(BadRequestError);
+  });
+
+  // attempt to promote a user to an admin
+  test('attempt to promote a user to an admin', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+
+    const staffUser = UserFactory.fake({ accessType: UserAccessType.STAFF });
+    const standardUser = UserFactory.fake({ accessType: UserAccessType.STANDARD });
+    const marketingUser = UserFactory.fake({ accessType: UserAccessType.MARKETING });
+    const merchStoreDistributorUser = UserFactory.fake({ accessType: UserAccessType.MERCH_STORE_DISTRIBUTOR });
+
+    await new PortalState()
+      .createUsers(staffUser, standardUser, marketingUser, merchStoreDistributorUser, admin)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        accessUpdates: [
+          { user: staffUser.email, accessType: UserAccessType.MERCH_STORE_MANAGER },
+          { user: standardUser.email, accessType: UserAccessType.ADMIN },
+          { user: marketingUser.email, accessType: UserAccessType.MERCH_STORE_DISTRIBUTOR },
+          { user: merchStoreDistributorUser.email, accessType: UserAccessType.STAFF },
+        ],
+      }, admin);
+    }).rejects.toThrow(ForbiddenError);
+  });
+
+  test('attempt to demote an existing admin', async () => {
```

Comment:
nit: this is expected to be successful, so "admin can demote another admin" would be more accurate.